spaceship/lib/spaceship/client.rb

```diff
@@ -446,12 +440,15 @@ def send_shared_login_request(user, password)
           modified_cookie.gsub!(unescaped_important_cookie, escaped_important_cookie)
         end
 
+        require 'uri'
+        encoded_password = URI.encode(password)
+        encoded_apple_id = URI.encode(user)
+
         response = request(:post) do |req|
-          req.url "https://idmsa.apple.com/appleauth/auth/signin"
-          req.body = data.to_json
-          req.headers['Content-Type'] = 'application/json'
-          req.headers['X-Requested-With'] = 'XMLHttpRequest'
-          req.headers['X-Apple-Widget-Key'] = self.itc_service_key
+          req.url "https://idmsa.apple.com/IDMSWebAuth/authenticate"
+          # appIdKey is hardcoded because that seems to the be an internal SSO service identifier.
+          req.body = "appIdKey=891bd3417a7776362562d2197f89480a8547b108fd934911bcbea0110d07f757&appleId=#{encoded_apple_id}&accountPassword=#{encoded_password}"
```

Comment:
for readability could we put the apiIdKey into a variable?

Reply:
I don't think we're actually allowed to have that API key as part of the fastlane code base